specification/resources/resource-manager/Microsoft.Management/2017-08-31-preview/management.json

```diff
@@ -0,0 +1,622 @@
+{
+ "swagger": "2.0",
+ "host": "management.azure.com",
+ "info": {
+  "version": "2017-08-31-preview",
+  "title": "Management Groups API",
+  "description": "The Azure Management Groups API enables consolidation of multiple \nsubscriptions/resources into an organizational hierarchy and centrally \nmanage access control, policies, alerting and reporting for those resources.\n"
+ },
+ "schemes": [
+  "https"
+ ],
+ "consumes": [
+  "application/json"
+ ],
+ "produces": [
+  "application/json"
+ ],
+ "security": [
+  {
+    "azure_auth": [
+      "user_impersonation"
+    ]
+  }
+],
+"securityDefinitions": {
+  "azure_auth": {
+    "type": "oauth2",
+    "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+    "flow": "implicit",
+    "description": "Azure Active Directory OAuth2 Flow.",
+    "scopes": {
+      "user_impersonation": "impersonate your user account"
+    }
+  }
+},
+ "tags": [
+  {
+   "name": "ManagementGroups",
+   "description": "A Management Group is a customer defined scope (grouping mechanism) that \nprovides access control (authorization), policy management and reporting. \nManagement Groups are organized in a strictly tree-based hierarchy.\n"
+  }
+ ],
+ "paths": {
+  "/providers/Microsoft.Management/managementGroups": {
+   "get": {
+    "tags": [
+     "ManagementGroups"
+    ],
+    "operationId": "ManagementGroups_List",
+    "description": "List management groups for the authenticated user.\n",
+    "parameters": [
+     {
+      "$ref": "#/parameters/ApiVersionParameter"
+     },
+     {
+      "$ref": "#/parameters/PageTokenParameter"
+     }
+    ],
+    "responses": {
+     "200": {
+      "description": "OK",
+      "schema": {
+       "$ref": "#/definitions/ManagementGroupListResult"
+      }
+     },
+     "default": {
+      "description": "Error",
+      "schema": {
+       "$ref": "#/definitions/ErrorResponse"
+      }
+     }
+    },
+    "x-ms-pageable": {
+     "nextLinkName": "nextLink"
```

Comment:
What exactly the content of `nextLink` property? 

is it (1) fully qualified url (2) relative url (3) something else (e.g. guid).. 

Asking because this operation also has a query parameter `PageTokenParameter` and it description says it will needs to be populated with the value of `nextLink` present in the current page in-order to get the next page.

Does your service requires user to follow below flow to enumerate all pages?

1. Get the first page `GET /providers/Microsoft.Management/managementGroups`. The response can contain value for `nextLink` e.g. abcd (just a token not a url)

2. Get the next page by using the original operation url `/providers/Microsoft.Management/managementGroups` and a query parameter `token` which is populated with the value of `nextLink` from the previous page

Reply:
"x-ms-pageable": {
          "nextLinkName": "nextLink"
        }

I am seeing same property defined in three other swaggers.
https://github.com/Azure/azure-rest-api-specs/blob/current/specification/billing/resource-manager/Microsoft.Billing/2017-04-24-preview/billing.json

I don't think there is anything wrong with this.

Reply:
@muazmian there is nothing wrong here w.r.t the usage of x-ms-pagable. My question is about relationship between `PageTokenParameter` and `nextLink` and how service is expecting user to use these properties together.

Reply:
@anuchandy ..The RP has not yet implemented anything regarding generating NextLink or PageTokenParameter but the plan is that nextLink will be a relative url with $skiptoken.

I am renaming the PageTokenParameter to SkipToken make it more clear.

Reply:
@muazmian, if an operation is marked as `x-ms-pagable` then the code generator emits two methods, in your case it generates following two methods:

```java
ManagementGroups {
 Page<ManagementGroupInfo>  List(String skipToken);
 Page<ManagementGroupInfo>  ListNext(String nextLinkUrl, String skipToken);
}
```

So user will use it as

```java
 Page<ManagementGroupInfo> firstPage = client.List( /* what should be value for skipToken param?*/);
 Page<ManagementGroupInfo> nextPage = client.ListNext(firstPage.nextLink, /* what should be value for skipToken param?*/);

```

 Iam trying to understand - If odata `$skipToken` is already present in the nextLink property when why do we need separate skipToken argument for the method?

Reply:
@anuchandy 

You are right, this is wrong. I have removed "x-ms-parameter-location" from $skipToken. Hopefully this will resolve the issue.

Reply:
In case this does not fix this, please let me know the actual fix. 

I am just helping the team in publishing the Swagger and Azure PS.

Reply:
removal of "x-ms-parameter-location" will simply make `skipTokenParameter` global, it goes to client from list method. Should we just remove this property given that it is not used anywhere?

Reply:
What do you think about moving the $SkipToken out of parameters and only define it for that 1 api that will use it? The reason I am asking is that so skipToken is shown as a query paramter on swagger

Reply:
sure sounds good..